### DISCUSSION TITLE
Why doesn't the rule work?

### DISCUSSION THREAD

**Discussion by uandrey1** [arendst/Tasmota#19800] [Q&A] [answered]
ESP32
tasmota32
I wrote a simple rule:
Rule1 ON PID#PidPower DO %value% ENDON
I enable the rule set, but rule doesn't work.
SENSOR = {"Time":"2023-10-19T16:06:49","COUNTER":{"C1":1038139,"C4":1082379},"DS18B20":{"Id":"031661E2D3FF","Temperature":25.4},"PID":{"PidPv":25.00,"PidSp":25.00,"PidPb":5.00,"PidTi":1800.00,"PidTd":15.00,"PidInitialInt":0.50,"PidDSmooth":3.00,"PidAuto":1,"PidManualPower":0.00,"PidMaxInterval":300,"PidInterval":300,"PidUpdateSecs":0,"PidSensorLost":0,"PidPower":0.48},"TempUnit":"C"}

If I write instead PID#PidPower DS18B20#Temperature with this sensor the rule works as expected. Is this some other type of sensor?

**Comment by sfromis** [comment]
May depend on what you mean by "doesn't work". Your rule does not make sense as in having no valid command to execute something, but should still result in a console message about `{"Command":"Unknown"}`. What do you see in the console when you expect it to trigger?

The trigger format `PID#PidPower` looks reasonable for that JSON, but for teleperiod, maybe also try `tele-PID#PidPower`.

**Reply by uandrey1** [reply]
I don't see any changes in the console with trigger PID#PidPower. But when the trigger is DS18B20#Temperature I see the result of this rule being triggered:
17:25:59.723 RUL: DS18B20#TEMPERATURE performs "22.9"
17:25:59.731 RSL: RESULT = {"Command":"Unknown"}
17:26:04.424 RUL: DS18B20#TEMPERATURE performs "22.9"
17:26:04.431 RSL: RESULT = {"Command":"Unknown"}
17:26:09.133 RUL: DS18B20#TEMPERATURE performs "23.0"
17:26:09.141 RSL: RESULT = {"Command":"Unknown"}
17:26:13.835 RUL: DS18B20#TEMPERATURE performs "23.0"
17:26:13.844 RSL: RESULT = {"Command":"Unknown"}
17:26:18.542 RUL: DS18B20#TEMPERATURE performs "23.0"
17:26:18.550 RSL: RESULT = {"Command":"Unknown"}
17:26:23.245 RUL: DS18B20#TEMPERATURE performs "23.0"
17:26:23.252 RSL: RESULT = {"Command":"Unknown"}
17:26:27.947 RUL: DS18B20#TEMPERATURE performs "23.0"
17:26:27.956 RSL: RESULT = {"Command":"Unknown"}

tele-PID#PidPower Triggered!!!
17:45:38.958 RUL: TELE-PID#PIDPOWER performs "0.31"
17:45:38.966 RSL: RESULT = {"Command":"Unknown"}

I can't understand why PID#PidPower doesn't trigger

**Reply by sfromis** [reply, marked as answer]
I don't know the code of the PID function, but it may only provide values at TelePeriod.

And do consider using valid commands.

**Reply by uandrey1** [reply]
Thank you!
